### PR TITLE
Fix sparse Merkle Tree test that randomly flaked occasionally.

### PR DESCRIPTION
**Pull request by Martin2112**
To reproduce flake in current code use: 
```
go test -count 100 -run TestInclusionProofGetsIncorrectNode github.com/google/trillian/merkle/...
```
Use of random data could generate nodes with short nodeIDs (8 bits) that then matched one of the siblings built from this ID, this broke the test. Collisions at longer node ID lengths were possible, but less likely.

I've made it deterministic now so this problem should not recur.

### CODE REVIEW

File: merkle/sparse_merkle_tree_test.go [hunk 1]
```diff
@@ -231,17 +241,21 @@ func TestInclusionProofGetsIncorrectNode(t *testing.T) {
...
+	// of the proof for the supplied key. This should not succeed.
+	for _, testNode := range inclusionProofIncorrectTestNodes {
+		const rev = 100
+		r, tx := getSparseMerkleTreeReaderWithMockTX(mockCtrl, rev)
+		tx.EXPECT().Commit().AnyTimes().Return(nil)
+		tx.EXPECT().GetMerkleNodes(int64(rev), gomock.Any()).Return([]storage.Node{testNode}, nil)
+		const key = "SomeArbitraryKey"
+		proof, err := r.InclusionProof(rev, []byte(key))
+		if err == nil {
+			t.Fatalf("InclusionProof() should've returned an error due to incorrect node from storage layer: %v", proof)
```
Comment (daviddrysdale) [inline merkle/sparse_merkle_tree_test.go]: As we're making changes could we fix the pre-existing: 
 - no need to be fatal
 - prefer canonical form for error messages.

Comment (Martin2112) [inline merkle/sparse_merkle_tree_test.go]: Done.

File: merkle/sparse_merkle_tree_test.go [hunk 2]
```diff
@@ -251,10 +251,10 @@ func TestInclusionProofGetsIncorrectNode(t *testing.T) {
 		const key = "SomeArbitraryKey"
 		proof, err := r.InclusionProof(rev, []byte(key))
 		if err == nil {
-			t.Fatalf("InclusionProof() should've returned an error due to incorrect node from storage layer: %v", proof)
+			t.Errorf("InclusionProof() = %v, nil. Want: nil, error", proof)
 		}
 		if !strings.Contains(err.Error(), "1 remain(s) unused") {
-			t.Fatalf("Saw unexpected error: %v", err)
+			t.Errorf("InclusionProof() = %v, %v. Want: 1 remain(s) unused", proof, err)
```
Comment (daviddrysdale) [inline merkle/sparse_merkle_tree_test.go]: Any reason not to make the two error messages more similar ("want: nil, '1 remain(s) unused'")?

Comment (Martin2112) [inline merkle/sparse_merkle_tree_test.go]: Done.

**Pull request merged**